Modules/_datetimemodule.c

```diff
@@ -1856,6 +1856,53 @@ make_freplacement(PyObject *object)
     return PyUnicode_FromString(freplacement);
 }
 
+#if defined(MS_WINDOWS) || defined(__ANDROID__)
+static PyObject *
+make_dash_replacement(PyObject *object, Py_UCS4 ch, PyObject *timetuple)
+{
+    PyObject *strftime = NULL;
+    PyObject *fmt_obj = NULL;
+    PyObject *res = NULL;
+    PyObject *stripped = NULL;
+
+    strftime = PyImport_ImportModuleAttrString("time", "strftime");
+    if (strftime == NULL) {
+        goto error;
+    }
+
+    fmt_obj = PyUnicode_FromFormat("%%%c", (char)ch);
+    if (fmt_obj == NULL) {
+        goto error;
+    }
+
+    res = PyObject_CallFunctionObjArgs(strftime, fmt_obj, timetuple, NULL);
+    if (res == NULL) {
+        goto error;
+    }
+
+    stripped = PyObject_CallMethod(res, "lstrip", "s", "0");
+    if (stripped == NULL) {
+        goto error;
+    }
+
+    if (PyUnicode_GET_LENGTH(stripped) == 0) {
+        stripped = PyUnicode_FromString("0");
+    }
```

Comment:
```suggestion
    if (PyUnicode_GET_LENGTH(stripped) == 0) {
        Py_DECREF(stripped);
        stripped = PyUnicode_FromString("0");
    }
```

There is a refleak (see buildbots), you need to decref the original stripped I think.